### PR TITLE
when shutdownArangod() already stopped arangods, don't look for the status again

### PR DESCRIPTION
**Pull request by dothebart**
this fixes the output showing up in some tests: `2018-07-23T08:14:05Z [8006] WARNING checkExternal: pid not found: 8164`

### CODE REVIEW

**Comment by dothebart** [comment]
A running Arangod that has never been `statusExternal()`'d looks like that:
```JSON
{ 
  "role" : "single", 
  "port" : 8958, 
  "endpoint" : "tcp://127.0.0.1:8958", 
  "rootDir" : "/tmp/arangosh_FzR7af/http_server", 
  "url" : "http://127.0.0.1:8958", 
  "pid" : 12082 
} 
```
Once `statusExternal()` was invoked once, but the process is still running:
```JSON
{ 
  "role" : "single", 
  "port" : 8958, 
  "endpoint" : "tcp://127.0.0.1:8958", 
  "rootDir" : "/tmp/arangosh_FzR7af/http_server", 
  "url" : "http://127.0.0.1:8958", 
  "pid" : 12082, 
  "exitStatus" : { 
    "status" : "RUNNING" 
  } 
}
```
On an already shutdown instance it looks like that:
```JSON
{ 
  "rootDir" : "/tmp/arangosh_Zw0IZe/endpoint-tcpv4", 
  "arangods" : [ 
    { 
      "role" : "single", 
      "port" : "4709", 
      "endpoint" : "tcp://127.0.0.1:4709", 
      "rootDir" : "/tmp/arangosh_Zw0IZe/endpoint-tcpv4", 
      "url" : "http://127.0.0.1:4709", 
      "pid" : 12216, 
      "exitStatus" : { 
        "pid" : 12216, 
        "status" : "TERMINATED", 
        "exit" : 0 
      } 
    } 
  ], 
  "protocol" : "tcp", 
  "endpoint" : "tcp://127.0.0.1:4709", 
  "url" : "http://127.0.0.1:4709" 
}
```

**Comment by dothebart** [comment]
Starting linux test only: http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/1237/

**Comment by dothebart** [comment]
Endpoint tests still red, error message gone. source for endpoint being bad will be continued in another PR.

**Comment by dothebart** [comment]
More code changes have been done to simplify the cluster shutdown sequence, so that there is just one loop with all the logic. http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/1263/

File: js/client/modules/@arangodb/crash-utils.js [hunk 1]
```diff
@@ -172,7 +172,7 @@ function analyzeCoreDumpWindows (instanceInfo) {
 // /        information about the incident.
 // //////////////////////////////////////////////////////////////////////////////
 function analyzeCrash (binary, arangod, options, checkStr) {
-  if (!options.coreCheck) {
+  if (!options.coreCheck || arangod.exitStatus.hasOwnProperty('gdbHint')) {
```
Comment (KVS85) [inline js/client/modules/@arangodb/crash-utils.js]: Will `arangod.exitStatus` present here in all cases of `analyzeCrash` usage?

Comment (dothebart) [reply js/client/modules/@arangodb/crash-utils.js]: yes, it should only invoke the crash analysis once it detected the errnous death of a server, or killed it by timeout with SIG_ABRT. This should stop the gdb run from occuring a second time.

**Pull request merged**